### PR TITLE
[interp] Outline `set_resume_state` to conserve stack in main interpreter loop.

### PR DESCRIPTION
**Pull request by jaykrell**
This is not intuitive but I definitely saw it.

### CODE REVIEW

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -229,7 +229,9 @@ int mono_interp_traceopt = 0;
 #define MINT_IN_DEFAULT default:
 #endif
 
-static void
+static
+MONO_NEVER_INLINE // Inling this causes caller to use more stack.
```
Comment (BrzVlad) [inline mono/mini/interp/interp.c]: single line if possible

Comment (jaykrell) [reply mono/mini/interp/interp.c]:
```
static MONO_NEVER_INLINE void
foo
```
ok, I can search for ^foo.


```
static MONO_NEVER_INLINE void foo
```
less ok but still ok

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/interp/interp.c [hunk 2]
```diff
@@ -229,7 +229,7 @@ int mono_interp_traceopt = 0;
 #define MINT_IN_DEFAULT default:
 #endif
 
-static void
+static MONO_NEVER_INLINE void // Inlining this causes caller to use more stack.
 set_resume_state (ThreadContext *context, InterpFrame *frame)
```
Comment (vargaz) [inline mono/mini/interp/interp.c]: The comment is kinda not neccessary, we do it for the same reason for all functions.

Comment (jaykrell) [reply mono/mini/interp/interp.c]: It isn't obvious here since there are no locals.
I guess it increases register pressure in caller.

Comment (jaykrell) [reply mono/mini/interp/interp.c]: But ok either way, I can remove it.

**Pull request merged**